File: C#/Problem 10 - Summation of primes.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Problem10{
  private static long GeneratePrime(){
    long cand=primes.Last();
    while(true){
      cand+=2;
      if(cand%5==0){
        continue;
      }
      long sqrtCand=(long)Math.Sqrt(cand)+1;
      bool ok=true;
      for(int i=0; primes[i]<sqrtCand; i++){
        if(cand/primes[i]*primes[i]==cand){
          ok=false;
          break;
        }
      }
      if(ok){
        if(cand<2000000){
          primes.Add(cand);
        }
        return cand;
      }
    }
  }
  private static List<long> primes=new List<long>{2,3,5};
  public static void Main(){
    while(GeneratePrime()<2000000){};
    Console.WriteLine(primes.Sum());
  }
}

File: C++/Problem 10 - Summation of primes.cpp
#include <iostream>
#include <vector>
#include <cmath>

struct Prime{
  int generatePrime(void){
    int cand=primes.back();
    int sqrt_cand=sqrt(cand)+1;
    while(true){
      cand+=2;
      bool ok=true;
      for(auto i=0; primes[i]<sqrt_cand; i++){
         int div=cand/primes[i];
         if(div*primes[i]){
           ok=false;
          break;
        }
      }
      if(ok){
        primes.push_back(cand);
        return cand;
      }
    }
  }
  std::vector<int> primes{2,3,5};
};

auto main()->int{
  auto sum=10LL;
  Prime primes;
  while(primes.generatePrime()<10){
    sum+=primes.primes.back();
  }
  std::cout << sum << std::endl;
  return 0;
}
